auslib/blobs/base.py

```diff
@@ -123,3 +123,11 @@ def getFooterXML(self, updateQuery, update_type, whitelistedDomains, specialForc
 
     def getInnerXML(self, updateQuery, update_type, whitelistedDomains, specialForceHosts):
         raise NotImplementedError()
+
+    def containsForbiddenDomain(self, product, whitelistedDomains):
+        raise NotImplementedError()
```

Comment:
I think you'll need to implement this method on all of the other blob types too, namely: SuperBlob, WhitelistBlob, and DesupportBlob. Otherwise I don't think we'll be able to use them anymore :).